conjure-java-core/src/integrationInput/java/com/palantir/product/StrictOneCollectionField.java

```diff
@@ -0,0 +1,150 @@
+import javax.annotation.Nullable;
+import javax.annotation.processing.Generated;
+
+@JsonDeserialize(builder = StrictOneCollectionField.DefaultBuilder.class)
+@Generated("com.palantir.conjure.java.types.BeanGenerator")
+public final class StrictOneCollectionField {
+    private final List<String> myList;
+
+    private int memoizedHashCode;
+
+    private StrictOneCollectionField(List<String> myList) {
+        validateFields(myList);
+        this.myList = Collections.unmodifiableList(myList);
+    }
+
+    @JsonProperty("myList")
+    public List<String> getMyList() {
+        return this.myList;
+    }
+
+    @Override
+    public boolean equals(@Nullable Object other) {
+        return this == other
+                || (other instanceof StrictOneCollectionField && equalTo((StrictOneCollectionField) other));
+    }
+
+    private boolean equalTo(StrictOneCollectionField other) {
+        if (this.memoizedHashCode != 0
+                && other.memoizedHashCode != 0
+                && this.memoizedHashCode != other.memoizedHashCode) {
+            return false;
+        }
+        return this.myList.equals(other.myList);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = memoizedHashCode;
+        if (result == 0) {
+            result = this.myList.hashCode();
+            memoizedHashCode = result;
+        }
+        return result;
+    }
+
+    @Override
+    public String toString() {
+        return "StrictOneCollectionField{myList: " + myList + '}';
+    }
+
+    public static StrictOneCollectionField of(List<String> myList) {
+        return builder().myList(myList).build();
+    }
+
+    private static void validateFields(List<String> myList) {
+        List<String> missingFields = null;
+        missingFields = addFieldIfMissing(missingFields, myList, "myList");
+        if (missingFields != null) {
+            throw new SafeIllegalArgumentException(
+                    "Some required fields have not been set", SafeArg.of("missingFields", missingFields));
+        }
+    }
+
+    private static List<String> addFieldIfMissing(List<String> prev, Object fieldValue, String fieldName) {
+        List<String> missingFields = prev;
+        if (fieldValue == null) {
+            if (missingFields == null) {
+                missingFields = new ArrayList<>(1);
+            }
+            missingFields.add(fieldName);
+        }
+        return missingFields;
+    }
+
+    public static MyListStageBuilder builder() {
+        return new DefaultBuilder();
+    }
+
+    public interface MyListStageBuilder {
+        Completed_StageBuilder myList(@Nonnull List<String> myList);
+
+        Builder from(StrictOneCollectionField other);
+    }
+
+    public interface Completed_StageBuilder {
+        StrictOneCollectionField build();
+    }
+
+    public interface Builder extends MyListStageBuilder, Completed_StageBuilder {
+        @Override
+        Builder myList(@Nonnull List<String> myList);
```

Comment:
There's an argument that this should take an `Iterable` parameter, and we're being too _strict_ with strict builders. I think it's ok to enforce that the provided argument is exactly the type defined in the definition. Let me know your thoughts.

Reply:
I don't think it matters what shape of iterable is provided, the important thing is that the data is set in some way (no reason to force folks to transform collection->list themselves)

Reply:
Fair. I think that tracks closer with "strict" meaning that every field is set.

Following the same reasoning, it sounds reasonable to add auxiliary methods, so long as the fields are set to something initially.

Reply:
Ok, after some more thought, it's probably best to keep these auxiliary methods. I think the optional value setter should be in the stage where the option is set though.

Will put up the diff to show exactly what I mean.

Reply:
Updated.